### PR TITLE
Document/unexpose anything we want in our public-supported API for 1.0/s...

### PR DESCRIPTION
**Pull request by jsuereth**
...bt 0.13.x.

Review by @havocp

### CODE REVIEW

File: ui-interface/src/main/scala/sbt/BackgroundRun.scala [hunk 1]
```diff
@@ -47,6 +47,7 @@ object SbtBackgroundRunPlugin extends AutoPlugin {
       Def.inputTask {
         val mainClass = mainClassTask.value getOrElse error("No main class detected.")
         UIKeys.jobService.value.runInBackgroundThread(Keys.resolvedScoped.value, { (logger, uiContext) =>
+          // TODO - Copy the classpath into some tmp directory so we don't immediately die if a recompile happens.
```
Comment (havocp) [inline ui-interface/src/main/scala/sbt/BackgroundRun.scala]: totally unrelated to the PR, no?

File: serialization/src/main/scala/sbt/serialization/CanToString.scala [hunk 2]
```diff
@@ -23,6 +35,17 @@ object CanToString {
     _.toString, {
       s: String => TypeExpression.parse(s)._1
     })
```
Comment (havocp) [inline serialization/src/main/scala/sbt/serialization/CanToString.scala]: these three seem kind of BS here - they should really be wherever we have the file/uri/typeexpression pickler

Comment (havocp) [inline serialization/src/main/scala/sbt/serialization/CanToString.scala]: ugh, but we have the picklers in a bogus CanToStringPickler trait - it's just implementation detail that we generate the picklers from CanToString. I think we should 1) move the file/uri/typeexpression picklers to a logical trait, not an implementation-detail-based trait, 2) make the canToString instances private implementation details of the picklers.

**Comment by havocp** [comment]
Shouldn't everything in the json package be private[serialization] ?

File: ui-interface/src/main/scala/sbt/BackgroundRun.scala (hunk 1 above)
Comment (jsuereth) [inline ui-interface/src/main/scala/sbt/BackgroundRun.scala]: Yeah, that one snuck in, just a comment for cleanup later.

File: serialization/src/main/scala/sbt/serialization/CanToString.scala (hunk 2 above)
Comment (jsuereth) [inline serialization/src/main/scala/sbt/serialization/CanToString.scala]: +1 from me.

**Comment by jsuereth** [comment]
The json package itself is a private/implementation detail.  I'll make it further private thugh.

**Comment by jsuereth** [comment]
@havocp / @eed3si9n The JSON format is now hidden inside the serialziaiton package.   All usage is exposed via SerializedValue and all tests are updated to reflect this.  It was actually only tests whcih used scala-pickling directly.

File: serialization/src/main/scala/sbt/serialization/pickler/JavaExtraPicklers.scala [hunk 3]
```diff
@@ -33,9 +54,9 @@ trait CanToStringPicklers extends PrimitivePicklers {
     }
   }
   implicit val typeExpressionPickler: SPickler[TypeExpression] with Unpickler[TypeExpression] =
-    canToStringPickler[TypeExpression](implicitly[FastTypeTag[TypeExpression]], implicitly[CanToString[TypeExpression]])
+    canToStringPickler[TypeExpression](FastTypeTag[TypeExpression], JavaExtraPicklers.typeExpressionCanToString)
```
Comment (havocp) [inline serialization/src/main/scala/sbt/serialization/pickler/JavaExtraPicklers.scala]: typeExpression isn't really a Java type ... the other two make sense...

**Comment by havocp** [comment]
maybe we need TypeExpressionPicklers, wdyt. otherwise lgtm.

**Comment by havocp** [comment]
LGTM.

**Comment by havocp** [comment]
Travis 2/3, merging.

**Pull request merged**